doc/man1/flux-jobs.adoc

```diff
@@ -81,6 +81,16 @@ nnodes:: job node count (if job ran / is running), empty string otherwise
 nnodes_hyphen:: same as nnodes, but '-' if job has not run yet / never ran
 ranks:: job ranks (if job ran / is running), empty string otherwise
 ranks_hyphen:: same as ranks, but '-' if job has not run yet / never ran
+success:: True of False if job completed successfully, empty string otherwise
+success_hyphen:: same as success, but '-' if job has not yet completed
+exception:: True of False if job had an exception, empty string otherwise
+exception_hyphen:: same as exception, but '-' instead of empty string
+exception_severity:: If exception True, the highest severity, empty string otherwise
```

Comment:
Random idea, but maybe using `.` for the objects nested within exception will feel more natural to users familar with OOP.
```suggestion
exception.severity:: If exception True, the highest severity, empty string otherwise
```

Reply:
I like the idea, but I don't think it'll work.  If a string format field is done like this in python, i.e. `{exception.severity}`, won't Python think `exception` is an object and severity is a field within it?

My test:

```
class A:
    def __init__(self):
        self.a = "a"
        self.b = "b"

class Foo:
    def __init__(self):
        self.A = A()


str="{0.A.b}"
o = Foo()
print(str.format(o))
```

I'm sure there are tricks to get around this, but dunno if its worth the energy?

Reply:
> If a string format field is done like this in python, i.e. {exception.severity}, won't Python think exception is an object and severity is a field within it?

I assumed that is what @SteVwonder was suggesting. Create an `ExceptionInfo` object and make `severity`, `type`, etc. attributes of that object. This is way is more flexible than the generating the underscore names by hand...

Reply:
> Create an ExceptionInfo object and make severity, type, etc. attributes of that object. This is way is more flexible than the generating the underscore names by hand...

Yeah, I like @grondo's suggestion.  TBH, I didn't realize that python interpreted the `.` in format strings and called `getattr`, so I originally was just suggesting a string literal change.  But even so, I think the `.` syntax has promise for flexibility and extensibility.

Reply:
> I assumed that is what @SteVwonder was suggesting. Create an ExceptionInfo object and make severity, type, etc. attributes of that object. This is way is more flexible than the generating the underscore names by hand...

Oh! I just thought he meant replacing the `_` with `.` naming wise.  Lemme look into this idea.

Reply:
Hmmm.  Fiddling with this tonight.  Right now the attribute `exception` returns true or false.   i.e. `JobInfoObject.exception` is a boolean.  If we make `exception` an object, Would users have to specify `exception.exception` (or some other word) to get the boolean value?  I know we can get the string representation by overriding `__repr__` and/or  `__str__`, but that's not ideal.

Reply:
>  I know we can get the string representation by overriding __repr__ and/or __str__

The JobException class would be local to `flux-jobs`, right?  If so, overriding `__str__` seems like a viable option to me.  Were there any particular downsides to that override that you were thinking about/trying to avoid?

> Would we want exception.exception as the ideal format option?

Maybe something like `exception.occurred`?

Reply:
> Maybe something like exception.occurred?

I agree, though I think eventually the JobInfo and ExceptionInfo classes might be useful outside of `flux-jobs`.

Rather than being useful as an output format field (I don't see a real use case for printing `True`/`False` in the output), it seems like `exception.type` or `exception.severity` would be the most useful in `flux-jobs` output. Something like `exception.occurred` would be more useful as a filtering option down the road.

I guess one thing I didn't think of is that every job will need to have an ExceptionInfo object instantiated even if it means there was no exception, o/w something like `{exception.occurred}` would throw some kind of ... exception??

That is a downside to this approach I guess, though I bet with python there's a clever way to handle it.

Reply:
> Were there any particular downsides to that override that you were thinking about/trying to avoid?

Well at some point (such as for a derived state), we'd want to actually do boolean checks on an exception to determine what derived state to output.  i.e. `JobInfo.exception == True`, which I don't think overriding `__str__` or `__repr__` would work.

> I guess one thing I didn't think of is that every job will need to have an ExceptionInfo object instantiated even if it means there was no exception, o/w something like {exception.occurred} would throw some kind of ... exception??

Yeah, I think so.  In addition, since `JobInfo` is initialed like this:

```
        #  Set defaults, then update with job-info.list response items:
        combined_dict = self.defaults.copy()
        combined_dict.update(info_resp)
```

We have to override `exception` if it was in the list response.

Reply:
> Well at some point (such as for a derived state), we'd want to actually do boolean checks on an exception to determine what derived state to output. i.e. JobInfo.exception == True, which I don't think overriding __str__ or __repr__ would work.

Wouldn't you check `job.exception.occurred`?

Reply:
> Wouldn't you check job.exception.occurred?

Now we would, yes.  I was referring to pre-ExceptionInfo class.  Sorry if that was confusing :P

Reply:
got it, sorry! :+1: